### PR TITLE
Mccalluc/add fields to datasetmanager index

### PR DESCRIPTION
**Pull request by mccalluc**
Towards #1815: I want facets that will work across datasets. I'm not sure who has the most experience here, so listing everyone.

The test needs to be stronger, and after that the next step would be to use it in the UI.

### CODE REVIEW

**Comment by mccalluc** [comment]
(Tests still pass, which indicates that we need better coverage here.)

File: refinery/data_set_manager/search_indexes.py [hunk 1]
```diff
@@ -60,33 +60,20 @@ def prepare(self, object):
...
+            value = annotation.attribute_value
+            if annotation.attribute_value_unit is not None:
+                value += " " + annotation.attribute_value_unit
+
+            name = re.sub(r'\W',
+                          settings.REFINERY_SOLR_SPACE_DYNAMIC_FIELDS,
+                          name)
+
+            uniq_key = name + "_" + uuid + "_s"
+            generic_key = name + "_generic_s"
```
Comment (mccalluc) [inline refinery/data_set_manager/search_indexes.py]: Adding "generic" fields to solr is the reason for this PR. Is this a reasonable approach?

File: refinery/data_set_manager/tests.py [hunk 2]
```diff
@@ -1572,3 +1571,62 @@ def test_get_basic_node(self):
...
+class NodeIndexTests(APITestCase):
+
+    def setUp(self):
+        investigation = Investigation.objects.create()
+        study = Study.objects.create(investigation=investigation)
+        assay = Assay.objects.create(study=study)
+
+        test_file = StringIO()
+        test_file.write('Coffee is great.\n')
+        file_store_item = FileStoreItem.objects.create(
```
Comment (scottx611x) [inline refinery/data_set_manager/tests.py]: I've found that something in our codebase is still writing temp files to disk for FileStoreItems regardless of the class you use for the `datafile` param

We have a pre_delete signal for FileStoreItems that deletes these datafiles, so I've been doing this in my tests:

```
def tearDown(self):
    FileStoreItem.objects.all().delete()
```

File: refinery/data_set_manager/search_indexes.py (hunk 1 above)
Comment (scottx611x) [inline refinery/data_set_manager/search_indexes.py]: @mccalluc I wouldn't know myself without doing more research about solr. 
I think @flekschas  would have more of an idea than any other team member.

**Comment by scottx611x** [comment]
@mccalluc Also regarding the travis test failures I found overriding the `maxDiff` attribute of TestCases to be helpful:
```
# Some members in assertions are truncated if they are too long, but we
# want to test them in their entirety
maxDiff = None
```

File: refinery/data_set_manager/tests.py [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
 from django.core.files.uploadedfile import (InMemoryUploadedFile,
                                             SimpleUploadedFile)
-
 from django.http import QueryDict
 from django.test import TestCase
-
 from rest_framework.test import APIClient, APIRequestFactory, APITestCase
 
-from .models import Assay, AttributeOrder, Study, Investigation, Node
+from core.models import DataSet, ExtendedGroup, InvestigationLink
```
Comment (jkmarx) [inline refinery/data_set_manager/tests.py]: code style guide: 
```Use explicit relative imports:
from __future__ import absolute_import

from .models import WaffleCone
from .forms import WaffleConeForm
from core.views import FoodMixin```

Comment (jkmarx) [inline refinery/data_set_manager/tests.py]: Which needs to be updated?

Comment (mccalluc) [inline refinery/data_set_manager/tests.py]: isort wants to put relative imports at the end, which doesn't seem bad to me, but I'll look into overriding that: #1830.

File: refinery/data_set_manager/search_indexes.py (hunk 1 above)
Comment (mccalluc) [inline refinery/data_set_manager/search_indexes.py]: Talked this over with Nils and he thinks its reasonable.

**Pull request merged**